### PR TITLE
Add CONTRIBUTING.md to solicit contributions

### LINKED ISSUE #845: Add CONTRIBUTING.md to provide others with a short guide to how they can help with the project
AC:
* Add info on ways to contribute
* Add details on how to set up dev environment
* Add info on how to submits PRs
* Add Code of Conduct to set the tone for contributions 
* Include a table of contents with links to document headings

### PR DESCRIPTION
**Pull request by uncommon-type**
## Description

This PR adds a `CONTRIBUTING.md` file  to provide potential contributors with a short guide on how to help with the project. It explains the ways to get involved as well as gives setup/installation details and information on how to submit Pull Requests. 

## Related Issue

Closes #845 

## Acceptance Criteria

- [x]  Info is included on ways to contribute
- [x]  Details are provided on how to set up dev environment
- [x]  Guidelines are given on how to submits pull requests
- [x] Table of contents is added with links to document headings

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|    | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|   | :link: Update dependencies |
|  ✓  | :scroll: Docs              |

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+
+
+## Ways to contribute
+
+There are various ways to get involved, here are a few suggestions of things we'd love some help with.
+
+#### Resolving existing issues
+
+ You can consider helping out with [issues](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues) already requiring attention.
```
Comment (ljharb) [inline CONTRIBUTING.md:29]: should this link to the "help wanted" label?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Reporting issues
+
+If you run into problems while using  ``eslint-plugin-jsx-a11y``, you can report them by opening a new issue  and providing as much context as you can about what you're running into along with a failing test or code sample to help others verify and ultimately fix the issue.
```
Comment (ljharb) [inline CONTRIBUTING.md:33]:
```suggestion
If you run into problems while using `eslint-plugin-jsx-a11y`, you can report them by opening a new issue and providing as much context as you can about what you're running into, along with a failing test or code sample to help others verify and ultimately fix the issue.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+> **Note:** Before submitting an issue, please check open and closed issues to see if the problem has already been reported.
```
Comment (ljharb) [inline CONTRIBUTING.md:35]:
```suggestion
> **Note:** Before filing an issue, please check both open and closed issues to see if the problem has already been reported.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+If you want to propose a new rule, please follow these [instructions](https://eslint.org/docs/developer-guide/contributing/new-rules).
```
Comment (ljharb) [inline CONTRIBUTING.md:37]: these are eslint core's guidelines - probably worth folks reading, but not necessarily the criteria we'd apply here.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Documentation
+
+We are happy to welcome contributions  from anyone willing to improve documentation by adding missing information or making it more consistent and coherent.
```
Comment (ljharb) [inline CONTRIBUTING.md:41]:
```suggestion
We are happy to welcome contributions from anyone willing to improve documentation by adding missing information or making it more consistent and coherent.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  ```
+  npm install
+  ```
+* Run the tests and the build
+  ```
+  npm t && npm run build
+  ```
+* After making changes, verify them and update your code if there’re any failing tests
+  ```
+  npm run test
```
Comment (ljharb) [inline CONTRIBUTING.md:62]:
```suggestion
  npm test && npm run build
  ```
* After making changes, verify them and update your code if there’re any failing tests
  ```
  npm run tests-only
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+Commit the changes to your branch
+
+```
+git add .
+git commit
+```
+
+Before  sending the pull request, make sure your code is running on the latest available code by rebasing onto the upstream source
```
Comment (ljharb) [inline CONTRIBUTING.md:93]:
```suggestion
Before sending the pull request, make sure your code is running on the latest available code by rebasing onto the upstream source
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+Push your changes
+
+```
+git push origin issue1234
+```
+
+Send the [pull request](https://docs.github.com/en/pull-requests), make requested changes, and get merged 
+
+
+
```
Comment (ljharb) [inline CONTRIBUTING.md:115]:
```suggestion
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+    * `feat`  either for a backwards-compatible enhancement or for a rule change that adds reported problems
+    * `fix!`  for a backwards-incompatible bug fix
+    * `feat!` for a backwards-incompatible enhancement or feature
+    * `docs`  for changes to documentation only
+    * `chore` for changes that aren't user-facing
+    * `build` for changes to build process only
+    * `refactor` a change that doesn't affect APIs or user experience
+    * `test` just changes to test files
+    * `ci` changes to our CI configuration files and scripts
+    * `perf`  a code change that improves performance
```
Comment (ljharb) [inline CONTRIBUTING.md:131]: these are angular's/conventional-commit's commit message conventions, and i don't use them. in particular, i'm philosophically opposed to "chore" - chores are bad, repo maintenance is good.

my commit conventions are undocumented but somewhat consistent. i suppose i should probably document them in a gist, and then we can link to that here.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+Longer description here if necessary
+
+Fixes #1234
+```
+
+
+## Code of Conduct
+
+We adhere to the [OpenJS Foundation Code of Conduct](https://github.com/openjs-foundation/cross-project-council/blob/main/CODE_OF_CONDUCT.md) and expect
```
Comment (ljharb) [inline CONTRIBUTING.md:149]: this is a great idea, but we don't actually adhere to that in this github org yet. we could add a CODE_OF_CONDUCT.md file to https://github.com/jsx-eslint/.github, though, and then we don't need to mention it here at all.

**Review by ljharb — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 9 above)
Comment (uncommon-type) [reply CONTRIBUTING.md:131]: I've seen those listed on the ESLint website, this is why I've included them here😄. Will remove them for now and wait for your gist.

File: CONTRIBUTING.md (hunk 4 above)
Comment (uncommon-type) [reply CONTRIBUTING.md]: thank you for pointing out, I removed the reference😄

**Comment by ljharb** [comment]
This is pending https://github.com/jsx-eslint/.github/pull/1.

**Review by ljharb — APPROVED** [review]
Thanks!

**Pull request merged**